lib/src/block_parser.dart

```diff
@@ -193,7 +193,19 @@ abstract class BlockSyntax {
   /// Gets whether or not [parser]'s current line should end the previous block.
   static bool isAtBlockEnd(BlockParser parser) {
     if (parser.isDone) return true;
-    return parser.blockSyntaxes.any((s) => s.canParse(parser) && s.canEndBlock);
+    return parser.blockSyntaxes.any((s) {
+      if (!s.canParse(parser)) return false;
+      if (s is ListSyntax) {
+        // An empty list cannot interrupt a paragraph. See
+        // https://spec.commonmark.org/0.29/#example-255.
+        // Ideally, [BlockSyntax.canEndBlock] should be changed to be a method
+        // which accepts a [BlockParser], but this would be a breaking change,
+        // so we're going with this temporarily.
+        var match = s.pattern.firstMatch(parser.current);
+        return match[7]?.isNotEmpty ?? false;
```

Comment:
Where does this `7` come from? It makes me nervous...

Reply:
Yeah we have a few of these around. Especially in the list items. I've added a comment. It would be good to clean up some day; remove some regex usage.